accord-core/src/main/java/accord/utils/CheckpointIntervalArray.java

```diff
@@ -216,4 +217,34 @@ else if ((ri & BIT29) != 0)
         forEachRange.accept(p1, p2, p3, p4, start, end);
         return end;
     }
+
+    @Inline
+    public <P1, P2, P3, P4> int forEachKey(Key key, IndexedQuadConsumer<P1, P2, P3, P4> forEachScanOrCheckpoint, IndexedRangeQuadConsumer<P1, P2, P3, P4> forEachRange, P1 p1, P2 p2, P3 p3, P4 p4, int minIndex)
```

Comment:
why `Key` vs `RoutableKey`?  I don't directly see `Key` class being used, so curious on the intent

Reply:
this is your code I copied forward, isn't it? I didn't look carefully at the signature here. You're right that this looks odd, I'll take a closer look

Reply:
It's your generic type parameter

Reply:
ah thanks, sorry!